### PR TITLE
Handle offline entities with baseVersion and trunkVersion

### PR DESCRIPTION
**Pull request by ktuite**
Possible backend solution for https://github.com/getodk/central/issues/669 

[XML form](https://drive.google.com/file/d/1G3rPaca3d34XpVgdaEvd_S1451EuEX4U/view?usp=sharing) that allows branch id / base version / trunk version to be set. 

#### Before submitting this PR, please make sure you have:

- [ ] run `make test` and confirmed all checks still pass OR confirm CircleCI build passes
- [ ] verified that any code from external sources are properly credited in comments or that everything is internally sourced

### CODE REVIEW

**Review by ktuite — COMMENTED** [review]
Interactive review

File: test/integration/api/offline-entities.js [hunk 1]
```diff
@@ -0,0 +1,595 @@
...
+        .expect(200)
+        .then(({ body }) => {
+          body.currentVersion.version.should.equal(4);
+          body.currentVersion.baseVersion.should.equal(3);
+          body.currentVersion.data.should.eql({ age: '22', status: 'departed', first_name: 'Johnny' });
+
+          body.currentVersion.branchId.should.equal(branchId);
+          body.currentVersion.branchBaseVersion.should.equal(3);
+          body.currentVersion.trunkVersion.should.equal(1);
+        });
```
Comment (ktuite) [inline test/integration/api/offline-entities.js:488]: could check audit log for `submission.reprocess` event

File: test/integration/api/offline-entities.js [hunk 2]
```diff
@@ -0,0 +1,595 @@
...
+          .replace('<status>new</status>', '<status>working</status>')
+        )
+        .set('Content-Type', 'application/xml')
+        .expect(200);
+
+      await exhaust(container);
+      let backlogCount = await container.oneFirst(sql`select count(*) from entity_submission_backlog`);
+      backlogCount.should.equal(1);
+
+      // Second submission contains update after create (middle of branhc)
```
Comment (ktuite) [inline test/integration/api/offline-entities.js:555]: spelling

**Review by ktuite — COMMENTED** [review]
interactive review

**Review by matthew-white — APPROVED** [review]

File: lib/worker/jobs.js [hunk 3]
```diff
@@ -18,6 +18,7 @@ const jobs = {
   'submission.update.version': [ require('./submission').submissionUpdateVersion, require('./entity').createOrUpdateEntityFromSubmission ],
 
   'submission.update': [ require('./entity').createOrUpdateEntityFromSubmission ],
+  'submission.reprocess': [ require('./entity').createOrUpdateEntityFromSubmission ],
```
[the remark this answers is not in the document]
Comment (ktuite) [reply lib/worker/jobs.js:21]: Sticking with the submission.reprocess event for ease of implementation for now. It's maybe a bit weird that the audit log gets used for job queuing here for something that isn't really user-triggered, but it's an easy way to reuse existing infrastructure.

**Pull request merged**